test/e2e/ipfs-companion.test.js

```diff
@@ -0,0 +1,176 @@
+import { Builder, By, Key } from 'selenium-webdriver'
+import { describe, it, before } from 'mocha'
+import { expect } from 'chai'
+import fs from 'fs'
+import chrome from 'selenium-webdriver/chrome.js'
+import firefox from 'selenium-webdriver/firefox.js'
+import { fail } from 'assert'
+
+function delay (ms) {
+  return new Promise(res => setTimeout(res, ms), _ => {})
+}
+
+function getExtension (browserName) {
+  const version = process.env.IPFS_COMPANION_VERSION || JSON.parse(fs.readFileSync('add-on/manifest.common.json')).version
+  return `build/ipfs_companion-${version}_${browserName}.zip`
+}
+
+async function openChromium (extension) {
+  console.info('Opening Chromium browser')
+  const options = new chrome.Options()
+  if (extension !== undefined) {
+    options.addExtensions(extension)
+  }
+  options.addArguments('--lang=en-GB,en-US')
+  if (process.env.TEST_HEADLESS === 'true') {
+    options.addArguments('--headless=chrome')
+  }
+  const builder = new Builder()
+    .forBrowser('chrome')
+    .setChromeOptions(options)
+  if (process.env.SELENIUM_REMOTE_CHROMIUM_URL !== undefined) {
+    console.info(`Using remote webdriver: ${process.env.SELENIUM_REMOTE_CHROMIUM_URL}`)
+    builder.usingServer(process.env.SELENIUM_REMOTE_CHROMIUM_URL)
+  }
+  console.info('Starting Chromium')
+  const browser = await builder.build()
+  await delay(5000) // waiting for the extension to load
```

Comment:
I'm not a fan of hardcoded delays, though I know they're sometimes necessary. We should use something like pRetry instead with some kind of backoff (increasing delays). i.e. we should check if the thing is ready and return success when it is. 

Otherwise, we're saying no matter what improvements happen in the future, all of this code will be at least `sumOfAllDelays + actualRuntime` long. For e2e tests that I imagine we want to run on all PRs in the future.. that's not ideal.

We used to have `window.ipfsCompanion`, but that was removed.. @lidel @whizzzkid are there any other ways to confirm that companion is loaded and ready?

One option: use pRetry with `findExtensionUrl`